sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -785,5 +785,195 @@ private Request CreateGetRevisionsRequest(SettingSelector selector, string pageL
 
             return request;
         }
+
+        /// <summary>
+        /// Locks an existing <see cref="ConfigurationSetting"/> in the configuration store.
+        /// </summary>
+        /// <param name="key">The primary identifier of a configuration setting.</param>
+        /// <param name="label">The value used to group configuration settings.</param>
+        /// <param name="etag">The value of an etag indicates the state of a configuration setting within a configuration store.
+        /// If it is specified, the configuration setting is only locked if etag value matches etag value in the configuration store.
+        /// If no etag value is passed in, then the setting is always locked.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        public virtual async Task<Response> LockAsync(string key, string label = default, ETag etag = default, CancellationToken cancellationToken = default)
```

Comment:
No overload that takes `ConfigurationSetting`?

Reply:
We have an open issue about overloads that I need to get resolved.  Once we close on it, I'll update or not, based on the outcome.